pattern/pattern.go

```diff
@@ -59,11 +60,16 @@ noopLoop:
 			break noopLoop
 		}
 	}
-	if !any { // short-cut without a string copy
+	if !any && mode&EntireString == 0 { // short-cut without a string copy
 		return pat, nil
 	}
 	closingBraces := []int{}
 	var buf bytes.Buffer
+	// Enable matching `\n` with `.` as globs match `\n`
+	buf.WriteString("(?s)")
```

Comment:
I think this means we can never take the `return pat, nil` shortcut above, unfortunately - the pattern `foo*bar` needs to be able to match the string `foo\nbar` with a newline between the two words when EntireString isn't set, so we want the resulting regular expression to be `(?s)foo.*bar` rather than just `foo.*bar`.

Reply:
I can't seem to craft a test case where the shortcut fails. Go's regexp does in general traverse newlines, the exception is that the special character `.` does not by default. In the shortcut case we are returning patterns which could never contain the `.` character, so we don't need to add the `s` flag, correct?

Reply:
Of course you're right :)